src/Catalyst.Core.Modules.Dfs.Tests/IntegrationTests/CoreApi/FileSystemApiTest.cs

```diff
@@ -0,0 +1,992 @@
+                var cts = new CancellationTokenSource(TimeSpan.FromSeconds(30));
+                var content = await a.UnixFsApi.ReadAllTextAsync(cid, cts.Token);
+                Assert.Equal("some content", content);
+            }
+        }
+
+        [Fact]
+        public async Task GetTar()
+        {
+            var temp = MakeTemp();
+            try
+            {
+                var dir = ipfs.UnixFsApi.AddDirectoryAsync(temp, true).Result;
+                var dirid = dir.Id.Encode();
+
+                var tar = await ipfs.UnixFsApi.GetAsync(dir.Id);
+                var archive = TarArchive.CreateInputTarArchive(tar);
+                var files = new List<string>();
+                archive.ProgressMessageEvent += (a, e, m) => { files.Add(e.Name); };
+                archive.ListContents();
+
+                Assert.Equal($"{dirid}", files[0]);
+                Assert.Equal($"{dirid}/alpha.txt", files[1]);
+                Assert.Equal($"{dirid}/beta.txt", files[2]);
+                Assert.Equal($"{dirid}/x", files[3]);
+                Assert.Equal($"{dirid}/x/x.txt", files[4]);
+                Assert.Equal($"{dirid}/x/y", files[5]);
+                Assert.Equal($"{dirid}/x/y/y.txt", files[6]);
+            }
+            finally
+            {
+                Directory.Delete(temp, true);
+            }
+        }
+
+        [Fact]
+        public async Task GetTar_RawLeaves()
+        {
+            var temp = MakeTemp();
+            try
+            {
+                var options = new AddFileOptions
+                {
+                    RawLeaves = true
+                };
+                var dir = ipfs.UnixFsApi.AddDirectoryAsync(temp, true, options).Result;
+                var dirid = dir.Id.Encode();
+
+                var tar = await ipfs.UnixFsApi.GetAsync(dir.Id);
+                var archive = TarArchive.CreateInputTarArchive(tar);
+                var files = new List<string>();
+                archive.ProgressMessageEvent += (a, e, m) => { files.Add(e.Name); };
+                archive.ListContents();
+
+                Assert.Equal($"{dirid}", files[0]);
+                Assert.Equal($"{dirid}/alpha.txt", files[1]);
+                Assert.Equal($"{dirid}/beta.txt", files[2]);
+                Assert.Equal($"{dirid}/x", files[3]);
+                Assert.Equal($"{dirid}/x/x.txt", files[4]);
+                Assert.Equal($"{dirid}/x/y", files[5]);
+                Assert.Equal($"{dirid}/x/y/y.txt", files[6]);
+            }
+            finally
+            {
+                Directory.Delete(temp, true);
+            }
+        }
+
+        [Fact]
+        public async Task GetTar_EmptyDirectory()
+        {
+            var temp = Path.Combine(Path.GetTempPath(), Path.GetRandomFileName());
+            Directory.CreateDirectory(temp);
+            try
+            {
+                var dir = ipfs.UnixFsApi.AddDirectoryAsync(temp, true).Result;
+                var dirid = dir.Id.Encode();
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove the unused local variable 'dirid'.](https://app.codacy.com/manual/catalyst-network/Catalyst.Framework/pullRequest?prid=4832783)